airflow/executors/base_executor.py

```diff
@@ -330,6 +337,30 @@ def trigger_tasks(self, open_slots: int) -> None:
         for _ in range(min((open_slots, len(self.queued_tasks)))):
             key, (command, _, queue, ti) = sorted_queue.pop(0)
 
+            if self.otel_use_context_propagation:
+                # If it's None, then the span for the current TaskInstanceKey hasn't been started.
+                if self.active_spans.get(key) is None:
```

Comment:
I've been chatting to @xBis7 about this, and we're not sure that this active_spans idea will work with Airflow's HA schedulers. We are exploring options.

Reply:
@ashb I think that the current approach covers all possible HA scenarios. Please let me know how it looks.